test/e2e/monitoring_stack_controller_test.go

```diff
@@ -517,6 +518,33 @@ func assertAlertmanagerReceivesAlerts(t *testing.T) {
 	}
 }
 
+func prometheusScaleDown(t *testing.T) {
+	numOfRep := int32(1)
+	ms := newMonitoringStack(t, "prometheus-scale-down-test")
+	ms.Spec.PrometheusConfig = &stack.PrometheusConfig{
+		Replicas: &numOfRep,
+	}
+
+	err := f.K8sClient.Create(context.Background(), ms)
+	assert.NilError(t, err, "failed to create a monitoring stack")
+
+	prom := monv1.Prometheus{}
+	f.GetResourceWithRetry(t, ms.Name, ms.Namespace, &prom)
+
+	assert.Equal(t, prom.Status.Replicas, int32(1))
+
+	key := types.NamespacedName{Name: ms.Name, Namespace: ms.Namespace}
+	err = f.K8sClient.Get(context.Background(), key, ms)
+	assert.NilError(t, err, "failed to get a monitoring stack")
+
+	numOfRep = 0
+	ms.Spec.PrometheusConfig.Replicas = &numOfRep
+	err = f.K8sClient.Update(context.Background(), ms)
+	assert.NilError(t, err, "failed to update a monitoring stack")
+	f.GetResourceWithRetry(t, ms.Name, ms.Namespace, &prom)
```

Comment:
shouldn't we retry if the status takes time to update and then fail?

Reply:
I assume that we want at least that generation == observedGeneration (for the MonitoringStack object) before retrieving the Prometheus object?